mylist.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'enumerable'
+class MyList
+  include MyEnumerable
+  def initialize(*args)
+    @list = args
+  end
+
+  def each
+    i = 0
+    while @list[i]
+      yield @list[i]
+      i += 1
+    end
+  end
```

Comment:
- [ ] Optional suggestions

You should return a normal Enumerator using to_enum, so I would recommend adding the below line before I = 0 declaration

- `return to_enum(:each) unless block_given?` and return @list after end of while.

Reply:
Thanks for the feedback